Crash when providing API key and secret from environment variable

I'd like to provide the API key and the secret as env variables rather than in the configs (.tf) files.
Maybe I'm doing something wrong, but terraform crashes. I've tried to re-init terraform, but it does not help.

1. Environment variables
```
[minibaby:~/tftest] mark% env | grep EXOSCALE
EXOSCALE_API_KEY=EXOfoo
EXOSCALE_SECRET_KEY=bar
```

2. Provider config (config.tf) with commented out values
```
provider "exoscale" {
  version = "~> 0.9"
  #key = "${var.apikey}"
  #secret = "${var.secret}"
}
```

3. Running terraform
```
[minibaby:~/tftest] mark% terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.template_file.init[1]: Refreshing state...
data.template_file.init[0]: Refreshing state...
data.template_cloudinit_config.xyz-test-lb-config[0]: Refreshing state...
data.template_cloudinit_config.xyz-test-lb-config[1]: Refreshing state...

------------------------------------------------------------------------

Error: Error running plan: 1 error(s) occurred:

* provider.exoscale: unexpected EOF

panic: runtime error: invalid memory address or nil pointer dereference
2018-06-01T13:39:08.350Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: [signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xaefe10]
... lots of stack trace messages ...
```

It looks as if it does not like the `provider "exoscale" {}` config lacking the `key` and `secret` element for some reason(?)

Terraform v0.11.7
+ provider.exoscale v0.9.22
+ provider.template v1.0.0

Thanks in advance!

I reran using TF_LOG=1, log output as follows.

Note: I'm running terraform from a docker image. Not sure if related with the crash, probably not but still telling you.

```
[minibaby:~/tftest] mark% docker run -i -t -v `pwd`:/app/ -w /app/ neoos-tf plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.template_file.init[0]: Refreshing state...
data.template_file.init[1]: Refreshing state...
data.template_cloudinit_config.rega-test-lb-config[1]: Refreshing state...
data.template_cloudinit_config.rega-test-lb-config[0]: Refreshing state...

Error: Error refreshing state: 1 error(s) occurred:

* provider.exoscale: unexpected EOF


panic: runtime error: invalid memory address or nil pointer dereference
2018-06-04T09:35:29.032Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: [signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xaefe10]
2018-06-04T09:35:29.032Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: 
2018-06-04T09:35:29.032Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: goroutine 14 [running]:
2018-06-04T09:35:29.032Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: github.com/exoscale/terraform-provider-exoscale/exoscale.providerConfigure(0xc4204a0070, 0x0, 0xc420422e00, 0xc4204a0070, 0x0)
2018-06-04T09:35:29.032Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: 	/home/yoan/exoscale/terraform-provider-exoscale/.gopath/src/github.com/exoscale/terraform-provider-exoscale/exoscale/provider.go:149 +0x390
2018-06-04T09:35:29.032Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: github.com/exoscale/terraform-provider-exoscale/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Configure(0xc4202cb880, 0xc42042acf0, 0xc420422dc0, 0xc420422de0)
2018-06-04T09:35:29.032Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: 	/home/yoan/exoscale/terraform-provider-exoscale/.gopath/src/github.com/exoscale/terraform-provider-exoscale/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:264 +0xdd
2018-06-04T09:35:29.033Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: github.com/exoscale/terraform-provider-exoscale/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Configure(0xc4204228c0, 0xc42042acf0, 0xc42000cb60, 0x0, 0x0)
2018-06-04T09:35:29.033Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: 	/home/yoan/exoscale/terraform-provider-exoscale/.gopath/src/github.com/exoscale/terraform-provider-exoscale/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:517 +0x3e
2018-06-04T09:35:29.033Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: reflect.Value.call(0xc42039e360, 0xc4203464c0, 0x13, 0xe30ccb, 0x4, 0xc420063f18, 0x3, 0x3, 0xc420350180, 0xc4204806e8, ...)
2018-06-04T09:35:29.033Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: 	/usr/lib/go/src/reflect/value.go:447 +0x969
2018-06-04T09:35:29.033Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: reflect.Value.Call(0xc42039e360, 0xc4203464c0, 0x13, 0xc420480718, 0x3, 0x3, 0xc4203405d0, 0xc400000003, 0x0)
2018-06-04T09:35:29.033Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: 	/usr/lib/go/src/reflect/value.go:308 +0xa4
2018-06-04T09:35:29.033Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: net/rpc.(*service).call(0xc42033d600, 0xc42033e5a0, 0xc42035e990, 0xc42035e9a0, 0xc420360800, 0xc4203433a0, 0xdae780, 0xc42042acf0, 0x16, 0xc7a140, ...)
2018-06-04T09:35:29.033Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: 	/usr/lib/go/src/net/rpc/server.go:384 +0x14e
2018-06-04T09:35:29.033Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: created by net/rpc.(*Server).ServeCodec
2018-06-04T09:35:29.033Z [DEBUG] plugin.terraform-provider-exoscale_v0.9.22_x1: 	/usr/lib/go/src/net/rpc/server.go:480 +0x43a
2018-06-04T09:35:29.035Z [DEBUG] plugin: plugin process exited: path=/tf-plugins/terraform-provider-exoscale_v0.9.22_x1
2018/06/04 09:35:29 [ERROR] root: eval: *terraform.EvalConfigProvider, err: unexpected EOF
2018/06/04 09:35:29 [ERROR] root: eval: *terraform.EvalSequence, err: unexpected EOF
2018/06/04 09:35:29 [ERROR] root: eval: *terraform.EvalOpFilter, err: unexpected EOF
2018/06/04 09:35:29 [ERROR] root: eval: *terraform.EvalSequence, err: unexpected EOF
2018/06/04 09:35:29 [TRACE] [walkRefresh] Exiting eval tree: provider.exoscale
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalIf
2018/06/04 09:35:29 [TRACE] root: eval: terraform.EvalNoop
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalGetProvider
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalReadDataDiff
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalIf
2018/06/04 09:35:29 [TRACE] root: eval: terraform.EvalNoop
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalGetProvider
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalReadDataDiff
2018/06/04 09:35:29 [TRACE] dag/walk: upstream errored, not walking "exoscale_affinity.rega-test-lb"
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalReadDataApply
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalReadDataApply
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalWriteState
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalUpdateStateHook
2018/06/04 09:35:29 [TRACE] [walkRefresh] Exiting eval tree: data.template_file.init[0]
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalWriteState
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalUpdateStateHook
2018/06/04 09:35:29 [TRACE] [walkRefresh] Exiting eval tree: data.template_file.init[1]
2018/06/04 09:35:29 [TRACE] dag/walk: walking "root"
2018/06/04 09:35:29 [TRACE] vertex 'root.root': walking
2018/06/04 09:35:29 [TRACE] dag/walk: walking "data.template_cloudinit_config.rega-test-lb-config"
2018/06/04 09:35:29 [TRACE] vertex 'root.data.template_cloudinit_config.rega-test-lb-config': walking
2018/06/04 09:35:29 [TRACE] vertex 'root.data.template_cloudinit_config.rega-test-lb-config': evaluating
2018/06/04 09:35:29 [TRACE] [walkRefresh] Entering eval tree: data.template_cloudinit_config.rega-test-lb-config
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalSequence
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalInterpolate
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalCountCheckComputed
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalIf
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalCountFixZeroOneBoundary
2018/06/04 09:35:29 [TRACE] [walkRefresh] Exiting eval tree: data.template_cloudinit_config.rega-test-lb-config
2018/06/04 09:35:29 [TRACE] vertex 'root.data.template_cloudinit_config.rega-test-lb-config': expanding/walking dynamic subgraph
2018/06/04 09:35:29 [TRACE] Graph after step *terraform.ResourceCountTransformer:

data.template_cloudinit_config.rega-test-lb-config[0] - *terraform.NodeRefreshableDataResourceInstance
data.template_cloudinit_config.rega-test-lb-config[1] - *terraform.NodeRefreshableDataResourceInstance
2018/06/04 09:35:29 [TRACE] OrphanResourceCount: Starting...
2018/06/04 09:35:29 [TRACE] OrphanResourceCount: Checking: data.template_cloudinit_config.rega-test-lb-config[0]
2018/06/04 09:35:29 [TRACE] OrphanResourceCount: Checking: data.template_cloudinit_config.rega-test-lb-config[1]
2018/06/04 09:35:29 [TRACE] Graph after step *terraform.OrphanResourceCountTransformer:

data.template_cloudinit_config.rega-test-lb-config[0] - *terraform.NodeRefreshableDataResourceInstance
data.template_cloudinit_config.rega-test-lb-config[1] - *terraform.NodeRefreshableDataResourceInstance
2018/06/04 09:35:29 [DEBUG] Attaching resource state to "data.template_cloudinit_config.rega-test-lb-config[1]": &terraform.ResourceState{Type:"template_cloudinit_config", Dependencies:[]string{"data.template_file.init.*"}, Primary:(*terraform.InstanceState)(0xc42040aa50), Deposed:[]*terraform.InstanceState{}, Provider:"provider.template", mu:sync.Mutex{state:0, sema:0x0}}
2018/06/04 09:35:29 [DEBUG] Attaching resource state to "data.template_cloudinit_config.rega-test-lb-config[0]": &terraform.ResourceState{Type:"template_cloudinit_config", Dependencies:[]string{"data.template_file.init.*"}, Primary:(*terraform.InstanceState)(0xc42040aa00), Deposed:[]*terraform.InstanceState{}, Provider:"provider.template", mu:sync.Mutex{state:0, sema:0x0}}
2018/06/04 09:35:29 [TRACE] Graph after step *terraform.AttachStateTransformer:

data.template_cloudinit_config.rega-test-lb-config[0] - *terraform.NodeRefreshableDataResourceInstance
data.template_cloudinit_config.rega-test-lb-config[1] - *terraform.NodeRefreshableDataResourceInstance
2018/06/04 09:35:29 [TRACE] Graph after step *terraform.TargetsTransformer:

data.template_cloudinit_config.rega-test-lb-config[0] - *terraform.NodeRefreshableDataResourceInstance
data.template_cloudinit_config.rega-test-lb-config[1] - *terraform.NodeRefreshableDataResourceInstance
2018/06/04 09:35:29 [DEBUG] ReferenceTransformer: "data.template_cloudinit_config.rega-test-lb-config[1]" references: []
2018/06/04 09:35:29 [DEBUG] ReferenceTransformer: "data.template_cloudinit_config.rega-test-lb-config[0]" references: []
2018/06/04 09:35:29 [TRACE] Graph after step *terraform.ReferenceTransformer:

data.template_cloudinit_config.rega-test-lb-config[0] - *terraform.NodeRefreshableDataResourceInstance
data.template_cloudinit_config.rega-test-lb-config[1] - *terraform.NodeRefreshableDataResourceInstance
2018/06/04 09:35:29 [TRACE] Graph after step *terraform.RootTransformer:

data.template_cloudinit_config.rega-test-lb-config[0] - *terraform.NodeRefreshableDataResourceInstance
data.template_cloudinit_config.rega-test-lb-config[1] - *terraform.NodeRefreshableDataResourceInstance
root - terraform.graphNodeRoot
  data.template_cloudinit_config.rega-test-lb-config[0] - *terraform.NodeRefreshableDataResourceInstance
  data.template_cloudinit_config.rega-test-lb-config[1] - *terraform.NodeRefreshableDataResourceInstance
2018/06/04 09:35:29 [TRACE] dag/walk: added new vertex: "data.template_cloudinit_config.rega-test-lb-config[0]"
2018/06/04 09:35:29 [TRACE] dag/walk: added new vertex: "data.template_cloudinit_config.rega-test-lb-config[1]"
2018/06/04 09:35:29 [TRACE] dag/walk: added new vertex: "root"
2018/06/04 09:35:29 [TRACE] dag/walk: added edge: "root" waiting on "data.template_cloudinit_config.rega-test-lb-config[0]"
2018/06/04 09:35:29 [TRACE] dag/walk: added edge: "root" waiting on "data.template_cloudinit_config.rega-test-lb-config[1]"
2018/06/04 09:35:29 [TRACE] dag/walk: dependencies changed for "root", sending new deps
2018/06/04 09:35:29 [TRACE] dag/walk: walking "data.template_cloudinit_config.rega-test-lb-config[0]"
2018/06/04 09:35:29 [TRACE] vertex 'root.data.template_cloudinit_config.rega-test-lb-config[0]': walking
2018/06/04 09:35:29 [TRACE] dag/walk: walking "data.template_cloudinit_config.rega-test-lb-config[1]"
2018/06/04 09:35:29 [TRACE] vertex 'root.data.template_cloudinit_config.rega-test-lb-config[1]': walking
2018/06/04 09:35:29 [TRACE] vertex 'root.data.template_cloudinit_config.rega-test-lb-config[0]': evaluating
2018/06/04 09:35:29 [TRACE] [walkRefresh] Entering eval tree: data.template_cloudinit_config.rega-test-lb-config[0]
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalSequence
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalWriteState
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalInterpolate
2018/06/04 09:35:29 [TRACE] vertex 'root.data.template_cloudinit_config.rega-test-lb-config[1]': evaluating
2018/06/04 09:35:29 [TRACE] [walkRefresh] Entering eval tree: data.template_cloudinit_config.rega-test-lb-config[1]
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalSequence
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalWriteState
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalInterpolate
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalIf
2018/06/04 09:35:29 [TRACE] root: eval: terraform.EvalNoop
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalGetProvider
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalReadDataDiff
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalIf
2018/06/04 09:35:29 [TRACE] root: eval: terraform.EvalNoop
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalGetProvider
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalReadDataDiff
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalReadDataApply
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalReadDataApply
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalWriteState
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalUpdateStateHook
2018/06/04 09:35:29 [TRACE] [walkRefresh] Exiting eval tree: data.template_cloudinit_config.rega-test-lb-config[1]
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalWriteState
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalUpdateStateHook
2018/06/04 09:35:29 [TRACE] [walkRefresh] Exiting eval tree: data.template_cloudinit_config.rega-test-lb-config[0]
2018/06/04 09:35:29 [TRACE] dag/walk: walking "root"
2018/06/04 09:35:29 [TRACE] vertex 'root.root': walking
2018/06/04 09:35:29 [TRACE] dag/walk: walking "provider.template (close)"
2018/06/04 09:35:29 [TRACE] vertex 'root.provider.template (close)': walking
2018/06/04 09:35:29 [TRACE] vertex 'root.provider.template (close)': evaluating
2018/06/04 09:35:29 [TRACE] dag/walk: upstream errored, not walking "exoscale_compute.rega-test-lb"
2018/06/04 09:35:29 [TRACE] dag/walk: upstream errored, not walking "provider.exoscale (close)"
2018/06/04 09:35:29 [TRACE] [walkRefresh] Entering eval tree: provider.template (close)
2018/06/04 09:35:29 [TRACE] root: eval: *terraform.EvalCloseProvider
2018/06/04 09:35:29 [TRACE] [walkRefresh] Exiting eval tree: provider.template (close)
2018/06/04 09:35:29 [TRACE] dag/walk: upstream errored, not walking "output.loadbalancer_ips"
2018/06/04 09:35:29 [TRACE] dag/walk: upstream errored, not walking "root"
2018/06/04 09:35:29 [DEBUG] plugin: waiting for all plugin processes to complete...
2018-06-04T09:35:29.053Z [WARN ] plugin: error closing client during Kill: err="connection is shut down"
2018-06-04T09:35:29.056Z [DEBUG] plugin: plugin process exited: path=/tf-plugins/terraform-provider-template_v1.0.0_x4



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

--- a/exoscale/provider.go
+++ b/exoscale/provider.go
@@ -134,9 +134,11 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 		profile := d.Get("profile").(string)
 
 		// Support `~/`
-		usr, _ := user.Current()
-		if strings.HasPrefix(config, "~/") {
-			config = filepath.Join(usr.HomeDir, config[2:])
+		usr, err := user.Current()
+		if err == nil {
+			if strings.HasPrefix(config, "~/") {
+				config = filepath.Join(usr.HomeDir, config[2:])
+			}
 		}
 
 		// Convert relative path to absolute

